app/controllers/application_controller.rb

```diff
@@ -1,10 +1,123 @@
 require './config/environment'
 
 class ApplicationController < Sinatra::Base
-
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions unless test?
+    set :session_secret, 'secret'
+  end
+
+  get '/' do
+    erb :index
+  end
+
+  get '/signup' do
+    redirect to '/tweets' if logged_in?
+    erb :'users/create_user'
+  end
+
+  get '/login' do
+    redirect to '/tweets' if logged_in?
+    erb :'/users/login'
+  end
+
+  get '/tweets' do
+    redirect to '/login' unless logged_in?
+    @user = current_user
+    erb :'/tweets/tweets'
+  end
+
+  get '/tweets/new' do
+    redirect to '/login' unless logged_in?
+    erb :'/tweets/new'
+  end
+
+  get '/tweets/:id' do
+    if logged_in?
+      @tweet = Tweet.find(params[:id])
+      erb :'/tweets/show_tweet'
+    else
+      redirect to '/login'
+    end
+  end
+
+  get '/tweets/:id/edit' do
+    if logged_in?
+      @tweet = Tweet.find_by(id: params[:id])
+      if @tweet.user == current_user
+        erb :'/tweets/edit_tweet'
+      end
+    else
+      redirect to '/login'
+    end
+  end
+
+  delete '/tweets/:id/delete' do
+    @tweet = Tweet.find(params[:id])
+    @tweet.delete if logged_in? && (@tweet.user == current_user)
+  end
+
+  get '/users/:slug' do
+    @user = User.find_by_slug(params[:slug])
+    erb :'/users/show'
+  end
+
+  get '/logout' do
+    session.clear if logged_in?
+    redirect to '/login'
+  end
+
+  post '/tweets/:id/edit' do
+    @tweet = Tweet.find(params[:id])
+    if logged_in? && (@tweet.user == current_user)
+      @tweet.update(content: params['content'])
+    else
+      redirect to '/login'
+    end
+  end
+
+  post '/tweets/new' do
```

Comment:
Typically, tweet creation would be posted to `/tweets` instead of `/tweets/new`.